### PR TITLE
Minor improvement to `qtbot` testing section

### PR DESCRIPTION
**Pull request by lucyleeow**
# References and relevant issues
Related zulip: https://napari.zulipchat.com/#narrow/stream/212875-general/topic/qtbot.20documentation

# Description

* Expand why you need QApplication running in test
* Recommend usage of qtbot 
* improve wording in comment

### CODE REVIEW

**Review by psobolewskiPhD — APPROVED** [review]

**Comment by lucyleeow** [comment]
Sorry @psobolewskiPhD I want to add some more items. Just didn't get around to it yet. I'll ping you once dine, thank you!

**Comment by lucyleeow** [comment]
@psobolewskiPhD okay I've tried to add more information but I am not knowledgeable on Qt so please let me know if anything is off!

File: docs/developers/contributing/testing.md [hunk 1]
```diff
@@ -205,38 +205,56 @@ you create during testing need to be cleaned up at the end of each test. We thus
 recommend that you use the following fixtures when needing a widget or
 {class}`~napari.Viewer` in a test.
 
-#### qtbot
+#### qapp and qtbot
 
-If you need a `QApplication` to be running for your test, you can use the
+If you need a `QApplication` to be running for your test you can use the
```
Comment (Czaki) [inline docs/developers/contributing/testing.md]:
```suggestion
If you need to use any Qt related code in your test you need to ensure that `QApplication` is created. To do this we suggest use  
```

File: docs/developers/contributing/testing.md [hunk 2]
```diff
@@ -205,38 +205,56 @@ you create during testing need to be cleaned up at the end of each test. We thus
...
+[`qapp`](https://pytest-qt.readthedocs.io/en/latest/reference.html#module-pytestqt.plugin)
+fixture from [`pytest-qt`](https://pytest-qt.readthedocs.io/en/latest/index.html),
+a napari testing dependency.
+
+If you need to instantiate a Qt object (e.g., a widget) for your test, we recommend
+that you use the
```
Comment (Czaki) [inline docs/developers/contributing/testing.md]:
```suggestion
If you need to instantiate a Qt GUI object (e.g., a widget) for your test, we recommend
that you use the
```

File: docs/developers/contributing/testing.md [hunk 3]
```diff
@@ -205,38 +205,56 @@ you create during testing need to be cleaned up at the end of each test. We thus
...
+@pytest.mark.usefixtures('qtbot')
+def test_something():
+    ...
+```
 <br/>
 ````
 
-`qtbot` provides a convenient
+`qtbot` also provides a convenient
 [`addWidget`](https://pytest-qt.readthedocs.io/en/latest/reference.html#pytestqt.qtbot.QtBot.addWidget)
```
Comment (Czaki) [inline docs/developers/contributing/testing.md]:
```suggestion
[`add_widget`/`addWidget`](https://pytest-qt.readthedocs.io/en/latest/reference.html#pytestqt.qtbot.QtBot.addWidget)
```

File: docs/developers/contributing/testing.md [hunk 4]
```diff
@@ -205,38 +205,56 @@ you create during testing need to be cleaned up at the end of each test. We thus
...
 method that will ensure that the widget gets closed at the end of the test.
-It *also* provides a whole bunch of other
-convenient methods for interacting with your GUI tests (clicking, waiting
-signals, etc...).  See the [`qtbot` docs](https://pytest-qt.readthedocs.io/en/latest/reference.html#pytestqt.qtbot.QtBot) for details.
+This prevents leakage between tests.
+The
+[`waitUntil`](https://pytest-qt.readthedocs.io/en/latest/reference.html#pytestqt.qtbot.QtBot.waitUntil) method is also useful to wait for a desired condition. The example below
```
Comment (Czaki) [inline docs/developers/contributing/testing.md]:
```suggestion
[`wait_until`/`waitUntil`](https://pytest-qt.readthedocs.io/en/latest/reference.html#pytestqt.qtbot.QtBot.waitUntil) method is also useful to wait for a desired condition. The example below
```

File: docs/developers/contributing/testing.md [hunk 5]
```diff
@@ -205,38 +205,56 @@ you create during testing need to be cleaned up at the end of each test. We thus
...
+def test_something():
+    ...
+```
 <br/>
 ````
 
-`qtbot` provides a convenient
+`qtbot` also provides a convenient
 [`addWidget`](https://pytest-qt.readthedocs.io/en/latest/reference.html#pytestqt.qtbot.QtBot.addWidget)
 method that will ensure that the widget gets closed at the end of the test.
```
Comment (Czaki) [inline docs/developers/contributing/testing.md]:
```suggestion
method that will ensure that the widget gets closed and properly cleaned at the end of the test. This prevents fropm random segfaults in tests run. 
```

File: docs/developers/contributing/testing.md (hunk 3 above)
Comment (lucyleeow) [reply docs/developers/contributing/testing.md]: Do you want both or is there one that you prefer? I guess `add_widget` would be more 'python-ic'?

Comment (lucyleeow) [reply docs/developers/contributing/testing.md]: Looks okay as both so will leave

**Comment by lucyleeow** [comment]
Thanks @Czaki ! Changes made

File: docs/developers/contributing/testing.md (hunk 3 above)
Comment (Czaki) [reply docs/developers/contributing/testing.md]: In our test we are using both of them. So I would like to signal that both are valid.

**Pull request merged**